### PR TITLE
snap: account for parallel installs in wrappers, place info and tests

### PR DESCRIPTION
**Pull request by bboozzoo**
Make sure that we pick up the right app names when handling apps from parallel
installations of the same snap.

### CODE REVIEW

**Comment by pedronis** [comment]
this seems a bit premature given that we haven't made the change to allow for that case, JoinSnapApp still does snap == app  (same issues with SplitSnapApp)

**Comment by bboozzoo** [comment]
@pedronis this started as a drive-by change, since then I've pushed a couple of patches that fix launcher command, app wrapper naming, minimal place info and extend the tests a bit. Please re-review at your convenience.

File: snap/info_test.go [hunk 1]
```diff
@@ -507,6 +516,10 @@ func (s *infoSuite) TestSplitSnapApp(c *C) {
 		{"foo.bar.baz", []string{"foo", "bar.baz"}},
 		// special case, snapName == appName
 		{"foo", []string{"foo", "foo"}},
+		// snap instance names
+		{"foo_instance.bar", []string{"foo_instance", "bar"}},
+		{"foo_instance.bar.baz", []string{"foo_instance", "bar.baz"}},
+		{"foo_instance", []string{"foo_instance", "foo"}},
```
Comment (chipaca) [inline snap/info_test.go]: 👍

File: snap/info_test.go [hunk 2]
```diff
@@ -523,6 +536,10 @@ func (s *infoSuite) TestJoinSnapApp(c *C) {
 		{[]string{"foo", "bar-baz"}, "foo.bar-baz"},
 		// special case, snapName == appName
 		{[]string{"foo", "foo"}, "foo"},
+		// snap instance names
+		{[]string{"foo_instance", "bar"}, "foo_instance.bar"},
+		{[]string{"foo_instance", "bar-baz"}, "foo_instance.bar-baz"},
+		{[]string{"foo_instance", "foo"}, "foo_instance"},
```
Comment (chipaca) [inline snap/info_test.go]: 👍

File: snap/info_test.go [hunk 3]
```diff
@@ -734,6 +779,13 @@ func (s *infoSuite) TestAppDesktopFile(c *C) {
 	c.Check(snapInfo.InstanceName(), Equals, "sample")
 	c.Check(snapInfo.Apps["app"].DesktopFile(), Matches, `.*/var/lib/snapd/desktop/applications/sample_app.desktop`)
 	c.Check(snapInfo.Apps["sample"].DesktopFile(), Matches, `.*/var/lib/snapd/desktop/applications/sample_sample.desktop`)
+
+	// snap with instance key
+	snapInfo.InstanceKey = "instance"
+	c.Check(snapInfo.InstanceName(), Equals, "sample_instance")
+	c.Check(snapInfo.Apps["app"].DesktopFile(), Matches, `.*/var/lib/snapd/desktop/applications/sample_instance_app.desktop`)
```
Comment (mvo5) [inline snap/info_test.go]: Not related to this PR but we could switch this code to use `..., Equals, dirs.GlobalRootDir + "/var/lib..."`

**Pull request merged**